fbw-a380x/src/wasm/systems/a380_systems/src/air_conditioning/mod.rs

```diff
@@ -493,27 +513,46 @@ impl A380AirConditioningSystem {
         pneumatic_overhead: &impl EngineBleedPushbutton<4>,
         pressurization_overhead: &A380PressurizationOverheadPanel,
     ) {
-        self.fdac.iter_mut().for_each(|controller| {
-            controller.update(
-                context,
-                &self.air_conditioning_overhead,
-                number_of_open_doors > 0,
-                engine_fire_push_buttons,
-                engines,
-                cpiom_b,
-                pneumatic,
-                pneumatic_overhead,
-                pressurization_overhead,
-            )
-        });
+        // CPIOM B1 and B3 calculate the LH AGU Flow Demand
+        // CPIOM B2 and B4 calculate the RH AGU Flow Demand
+        self.fdac
+            .iter_mut()
+            .enumerate()
+            .for_each(|(id, controller)| {
+                controller.update(
+                    context,
+                    &self.air_conditioning_overhead,
+                    number_of_open_doors > 0,
+                    engine_fire_push_buttons,
+                    engines,
+                    if !cpiom_b[id].ags_has_fault() {
+                        &cpiom_b[id]
+                    } else {
+                        &cpiom_b[id + 2]
+                    },
+                    pneumatic,
+                    pneumatic_overhead,
+                    pressurization_overhead,
+                );
+            });
+
+        let cpiom_to_use = if cpiom_b[0].tcs_has_fault() {
+            &cpiom_b[0]
+        } else if cpiom_b[1].tcs_has_fault() {
+            &cpiom_b[1]
+        } else if cpiom_b[2].tcs_has_fault() {
+            &cpiom_b[2]
+        } else {
+            &cpiom_b[3]
+        };
```

Comment:
Are you sure about these conditions? IMO this would always select the faulted one.

Reply:
This was definitely wrong 😅 Good spot thanks.